v0/destinations/splitio/transform.js

```diff
@@ -0,0 +1,120 @@
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  KEY_CHECK_LIST
+} = require("./config");
+const { EventType } = require("../../../constants");
+const {
+  removeUndefinedAndNullValues,
+  defaultPostRequestConfig,
+  defaultRequestConfig,
+  constructPayload
+} = require("../../util");
+
+let bufferProperty = {};
+
+function responseBuilderSimple(payload, category, destination) {
+  if (payload) {
+    const responseBody = payload;
+    const response = defaultRequestConfig();
+    response.endpoint = category.endPoint;
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.headers = {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${destination.Config.apiKey}`
+    };
+    response.body.JSON = removeUndefinedAndNullValues(responseBody);
+    return response;
+  }
+  // fail-safety for developer error
+  throw new Error("Payload could not be constructed");
+}
+
+function populateOutputProperty(inputObject) {
+  Object.keys(inputObject).forEach(key => {
```

Comment:
first check inputObject present or not then proceed

Reply:
we are checking if traits or context.traits is present before calling the function, do we need to check it inside it as well then?

Reply:
for track page screen calls you did not check..